overlord/ifacestate/ifacestate.go

```diff
@@ -56,9 +56,24 @@ func checkConnectConflicts(st *state.State, change *state.Change, plugSnap, slot
 	if autoConnectTask != nil {
 		snapsup, err := snapstate.TaskSnapSetup(autoConnectTask)
 		if err != nil {
-			return fmt.Errorf("internal error: cannot obtain snap setup from task: %s", autoConnectTask.Summary())
+			return false, fmt.Errorf("internal error: cannot obtain snap setup from task: %s", autoConnectTask.Summary())
 		}
 		installedSnap = snapsup.Name()
+
+		// if we find any auto-connect task that's not ready and is affecting our snap, return true to indicate that
+		// it should be ignored (we shouldn't create connect tasks for it)
+		for _, task := range st.Tasks() {
```

Comment:
given that we loop twice anyway in this formulation, wondering if it should be a separate function, it would also avoid the "false, " everywhere else

Reply:
Good idea, done. Note, I made one of the checks stricter there.